go/stats/main.go

```diff
@@ -334,19 +334,31 @@ func main() {
 	var keyConfStats pfconfigdriver.PfconfigKeys
 	keyConfStats.PfconfigNS = "config::Stats"
 	pfconfigdriver.FetchDecodeSocket(ctx, &keyConfStats)
+    re_metric := regexp.MustCompile("^metric .*")
 
 	for _, key := range keyConfStats.Keys {
 		var ConfStat pfconfigdriver.PfStats
 		ConfStat.PfconfigHashNS = key
 
 		pfconfigdriver.FetchDecodeSocket(ctx, &ConfStat)
-		files = append(files, ConfStat.File)
 
-		go tailFile(ConfStat, config, done)
+		if re_metric.MatchString(key) {
+			err = ProcessMetricConfig(ctx, ConfStat)
+			if err != nil {
+				log.LoggerWContext(ctx).Error(err.Error())
+			}
+		}
+
+		if ConfStat.File != "" {
```

Comment:
I'd move this to ProcessMetricConfig with the type being `file`

Reply:
Conditional was added here to address to fix runtime error in #3017 and has nothing to do with [metric .*] pfconfigdriver keys from `conf/stats.conf`

Reply:
Since we haven't started seriously to configure metrics and @fdurand left none in the example file, then I believe its safe to migrate this to the "mini-framework" you created